pkg/etcd/etcd.go

```diff
@@ -0,0 +1,81 @@
+package etcd
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const etcdNamespace = "openshift-etcd"
+
+var log = logf.Log.WithName("etcd-pdb-checker")
+
+// isEtcdDisruptionAllowed checks if etcd disruption is allowed and refuse the todoAction (remediation/manitenance) when it isn't allowed
+func isEtcdDisruptionAllowed(ctx context.Context, cl client.Client, todoAction string) (bool, *policyv1.PodDisruptionBudget, error) {
+	pdbList := &policyv1.PodDisruptionBudgetList{}
+	if err := cl.List(ctx, pdbList, &client.ListOptions{Namespace: etcdNamespace}); err != nil {
+		return false, nil, err
+	}
+	if len(pdbList.Items) == 0 {
+		log.Info("No PDB found, can't check if etcd quorum will be violated! Refusing "+todoAction+"!", "namespace", etcdNamespace)
+		return false, nil, nil
+	}
+	if len(pdbList.Items) > 1 {
+		log.Info("More than one PDB found, can't check if etcd quorum will be violated! Refusing "+todoAction+"!", "namespace", etcdNamespace)
+		return false, nil, nil
+	}
+	pdb := pdbList.Items[0]
+	if pdb.Status.DisruptionsAllowed >= 1 {
+		return true, &pdb, nil
+	}
+	return false, &pdb, nil
+}
+
+// IsControlPlaneNodeReady checks if etcd disruption is allowed and accpet/refuse the todoAction (remediation/manitenance)
```

Comment:
please add a comment about which permissions this needs

```
// +kubebuilder:rbac:groups=policy,resources=poddisruptionbudgets,verbs=get;list;watch
// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
```

Reply:
But if we add it here will the other project that calls for the `IsControlPlaneNodeReady` function will use these Kubebuilder lines to update RBAC?
Should these lines be written locally in the parent project or is it needed also here?

Reply:
I don't think kubebuilder looks at vendored code, so it won't have any effect. What I mean is that we should document that the parent project needs to add these lines

Reply:
>I don't think kubebuilder looks at vendored code, so it won't have any effect.

+1

> What I mean is that we should document that the parent project needs to add these lines

SGTM